### PR TITLE
Add support for ECDSA over P256 using NSS

### PR DESCRIPTION
**Pull request by jschanck**
This adds `crypto::backend` functions for generating P256 keys and signing with ECDSA P256 SHA256.

The "Add Credential" function in the Virtual Authenticator API (https://w3c.github.io/webauthn/#sctn-automation-add-credential) provides the private key as an RFC 5958 `PrivateKeyInfo`. I've made `ecdsa_p256_sha256_sign_raw` a public function within the crypto module since it already acts on `PrivateKeyInfo`s. If we start signing outside of tests, then we'll want a richer type for wrapping private keys. But we'll probably use `EncryptedPrivateKeyInfo` instead, so I'm going to defer locking in those details.

Depends on #278 and https://github.com/mozilla/nss-gk-api/pull/8.

### CODE REVIEW

File: src/crypto/nss.rs [hunk 1]
```diff
@@ -29,6 +34,118 @@ impl From<nss_gk_api::Error> for CryptoError {
...
+fn der_expect_tag_with_short_len(tag: u8, z: &[u8]) -> Result<(&[u8], &[u8])> {
+    if z.is_empty() {
+        return Err(CryptoError::MalformedInput);
+    }
+    let (h, z) = z.split_at(1);
+    if h[0] != tag || z.is_empty() {
+        return Err(CryptoError::MalformedInput);
+    }
+    let (h, z) = z.split_at(1);
+    if z.len() >= 0x80 || h[0] as usize > z.len() {
```
Comment (mozkeeler) [inline src/crypto/nss.rs]: Why is it important to enforce that `z.len() < 0x80`? `z` includes `value` and `rest`, so this puts a constraint on `rest` as well, which seems outside the scope of this function.

File: src/crypto/nss.rs [hunk 2]
```diff
@@ -395,3 +572,28 @@ pub fn test_ecdh_p256_raw(
...
+#[cfg(test)]
+pub fn test_ecdsa_p256_sha256_verify_raw(
+    public: &[u8],
+    signature: &[u8],
+    data: &[u8],
+) -> Result<()> {
+    nss_gk_api::init();
+
+    let signature = decode_der_p256_sig(signature)?;
+    println!("{:?}", data);
```
Comment (mozkeeler) [inline src/crypto/nss.rs]: Probably don't want this here?

**Review by mozkeeler — APPROVED** [review]
Just a couple of comments. LGTM, though.

File: src/crypto/nss.rs (hunk 1 above)
Comment (jschanck) [reply src/crypto/nss.rs]: good catch!
